R/rstar.R

```diff
@@ -0,0 +1,193 @@
+#'   default for gradient-based model (`method="gbm"`) is `interaction.depth=3,
+#'   n.trees = 50, shrinkage=0.1, n.minobsinnode=10`.
+#'
+#' @param nsimulations Number of R* values in the returned vector if
+#'   `uncertainty` is `TRUE`. Default is 1000.
+#'
+#' @param training_proportion Proportion of iterations used to train the
+#'   classifier. Default is 0.7.
+#'
+#' @details The `rstar()` function provides a measure of MCMC convergence based
+#'   on whether it is possible to determine the chain that generated a
+#'   particular draw with a probability greater than chance. To do so, it fits a
+#'   machine learning classifier to a subset of the original MCMC draws (the
+#'   training set) and evaluates its predictive accuracy on the remaining draws
+#'   (the testing set). If predictive accuracy exceeds chance (i.e. predicting
+#'   the chain that generated a draw uniformly at random), the diagnostic
+#'   measure R* will be above 1, indicating that convergence has yet to occur.
+#'   This statistic is recently developed, and it is currently unclear what is a
+#'   reasonable threshold for diagnosing convergence.
+#'
+#'   The statistic, R*, is stochastic, meaning that each time the test is run,
+#'   unless the random seed is fixed, it will generally produce a different
+#'   result. To minimise the implications of this stochasticity, it is
+#'   recommended to repeatedly run this function to calculate a distribution of
+#'   R*; alternatively, an approximation to this distribution can be obtained by
+#'   setting `uncertainty = TRUE`, although this approximation of uncertainty
+#'   will generally have a lower mean.
+#'
+#'   By default, a random forest classifier is used (`method="rf"`), which tends
+#'   to perform best for target distributions of around 4 dimensions and above.
+#'   For lower dimensional targets, gradient boosted models (called via
+#'   `method="gbm"`) tend to have a higher classification accuracy. On a given
+#'   MCMC sample, it is recommended to try both of these classifiers.
+#'
+#' @return A numeric vector of length 1 (by default) or length `nsimulations`
+#'   (if `uncertainty=TRUE`).
+#'
+#' @references Ben Lambert, Aki Vehtari (2020) R*: A robust MCMC convergence
+#'   diagnostic with uncertainty using gradient-boosted machines \emph{arXiv
+#'   preprint} \code{arXiv:TBD}.
+#'
+#' @examples
+#' x <- example_draws("eight_schools")
+#' rstar(x)
+#' rstar(x, split_chains = FALSE)
+#' rstar(x, method = "gbm")
+#' # can pass additional arguments to methods
+#' rstar(x, method = "gbm", verbose = FALSE)
+#'
+#' # with uncertainty, returns a vector of R* values
+#' hist(rstar(x, uncertainty = TRUE))
+#' hist(rstar(x, uncertainty = TRUE, nsimulations = 100))
+#'
+#' # can use other classification methods in caret library
+#' rstar(x, method = "knn")
+#'
+#' @export
+rstar <-
+  function(x,
+           split_chains = TRUE,
+           uncertainty = FALSE,
+           method = NULL,
```

Comment:
Is there a reason for `method=NULL` and not `method="rf"`?

Reply:
Nope. Changed.